src/connectors/index.ts

```diff
@@ -1,49 +1,59 @@
+import { GnosisSafe } from '@web3-react/gnosis-safe'
+import { MetaMask } from '@web3-react/metamask'
+import { Network } from '@web3-react/network'
+import { WalletConnect } from '@web3-react/walletconnect'
 import { ALL_SUPPORTED_CHAIN_IDS, SupportedChainId } from 'constants/chains'
 import { INFURA_NETWORK_URLS } from 'constants/infura'
-import { InjectedConnector } from 'web3-react-injected-connector'
-import { WalletConnectConnector } from 'web3-react-walletconnect-connector'
-import { WalletLinkConnector } from 'web3-react-walletlink-connector'
 
 import UNISWAP_LOGO_URL from '../assets/svg/logo.svg'
-import getLibrary from '../utils/getLibrary'
-import { FortmaticConnector } from './Fortmatic'
-import { NetworkConnector } from './NetworkConnector'
 
-const FORMATIC_KEY = process.env.REACT_APP_FORTMATIC_KEY
-
-export const network = new NetworkConnector({
-  urls: INFURA_NETWORK_URLS,
-  defaultChainId: 1,
-})
-
-let networkLibrary: Web3Provider | undefined
-export function getNetworkLibrary(): Web3Provider {
-  return (networkLibrary = networkLibrary ?? getLibrary(network.provider))
-}
-
-export const injected = new InjectedConnector({
-  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
-})
-
-export const gnosisSafe = new SafeAppConnector()
-
-export const walletconnect = new WalletConnectConnector({
-  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
-  rpc: INFURA_NETWORK_URLS,
-  qrcode: true,
-})
-
-// mainnet only
-export const fortmatic = new FortmaticConnector({
-  apiKey: FORMATIC_KEY ?? '',
-  chainId: 1,
-})
-
-export const walletlink = new WalletLinkConnector({
-  url: INFURA_NETWORK_URLS[SupportedChainId.MAINNET],
-  appName: 'Uniswap',
-  appLogoUrl: UNISWAP_LOGO_URL,
-  supportedChainIds: ALL_SUPPORTED_CHAIN_IDS,
-})
+export const [network, networkHooks] = initializeConnector<Network>(
+  (actions) => new Network(actions, INFURA_NETWORK_URLS, true, 1),
+  Object.keys(INFURA_NETWORK_URLS).map((chainId) => Number(chainId))
+)
+
+export const [injected, injectedHooks] = initializeConnector<MetaMask>(
+  (actions) => new MetaMask(actions, true),
+  ALL_SUPPORTED_CHAIN_IDS
+)
+
+export const [gnosisSafe, gnosisSafeHooks] = initializeConnector<GnosisSafe>((actions) => new GnosisSafe(actions, true))
+
+export const [walletConnect, walletConnectHooks] = initializeConnector<WalletConnect>(
+  (actions) =>
+    new WalletConnect(
+      actions,
+      {
+        rpc: INFURA_NETWORK_URLS,
+        qrcode: true,
+      },
+      true
+    ),
+  ALL_SUPPORTED_CHAIN_IDS
+)
+
+export const [coinbaseWallet, coinbaseWalletHooks] = initializeConnector<CoinbaseWallet>(
+  (actions) =>
+    new CoinbaseWallet(
+      actions,
+      {
+        url: INFURA_NETWORK_URLS[SupportedChainId.MAINNET],
+        appName: 'Uniswap',
+        appLogoUrl: UNISWAP_LOGO_URL,
+      },
+      true
+    ),
+  ALL_SUPPORTED_CHAIN_IDS
+)
+
+// this is an ordered priority list. network connector should be at the top because we want to always
```

Comment:
nits:
- Use full sentences (eg capitalize).
- "list ordered by priority" instead of "ordered priority list", to avoid a reader mistaking it for a data structure (like a "priority queue")